src/Microsoft.AspNetCore.OData/Routing/Controllers/ODataController.cs

```diff
@@ -52,5 +53,126 @@ protected virtual UpdatedODataResult<TEntity> Updated<TEntity>(TEntity entity)
+        {
+            return new UnauthorizedODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Unauthorized (401) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>An <see cref="UnauthorizedODataResult"/> with the specified values.</returns>
+        protected virtual UnauthorizedODataResult Unauthorized(ODataError odataError)
+        {
+            return new UnauthorizedODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Conflict (409) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>A <see cref="ConflictODataResult"/> with the specified values.</returns>
+        protected virtual ConflictODataResult Conflict(string message)
+        {
+            return new ConflictODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce a Conflict (409) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>A <see cref="ConflictODataResult"/> with the specified values.</returns>
+        protected virtual ConflictODataResult Conflict(ODataError odataError)
+        {
+            return new ConflictODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce an UnprocessableEntity (422) response.
+        /// </summary>
+        /// <param name="message">Error Message</param>
+        /// <returns>An <see cref="UnprocessableEntityODataResult"/> with the specified values.</returns>
+        protected virtual UnprocessableEntityODataResult UnprocessableEntity(string message)
+        {
+            return new UnprocessableEntityODataResult(message);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="StatusCodeResult"/> that when executed will produce an UnprocessableEntity (422) response.
+        /// </summary>
+        /// <param name="odataError">Parameter of type <see cref="ODataError"/>.</param>
+        /// <returns>An <see cref="UnprocessableEntityODataResult"/> with the specified values.</returns>
+        protected virtual UnprocessableEntityODataResult UnprocessableEntity(ODataError odataError)
+        {
+            return new UnprocessableEntityODataResult(odataError);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="ActionResult"/> that when executed will produce an <see cref="ODataError"/> response.
+        /// </summary>
+        /// <param name="errorCode">Http Error code.</param>
+        /// <param name="message">Http Error Message.</param>
```

Comment:
```suggestion
        /// <param name="message">Http error message.</param>
```
Though I'm not sure we should have "Http" here